### ISSUE TITLE
NRE spam in RSE_PartAudioManager.LateUpdate (loading saves?)

### ISSUE THREAD

**Issue by JonnyOThan** [KSPModStewards/RocketSoundEnhancement#33]
```
[EXC 11:01:11.910] NullReferenceException
    RocketSoundEnhancement.RSE_PartAudioManager.LateUpdate () (at <80da5f92e59a4e1e8c62d932c2676817>:0)
    UnityEngine.DebugLogHandler:LogException(Exception, Object)
    ModuleManager.UnityLogHandle.InterceptLogHandler:LogException(Exception, Object)
    UnityEngine.Debug:CallOverridenDebugHandler(Exception, Object)
```

Noticed this from TPKSP, possibly after loading a quicksave

[KSP.zip](https://github.com/user-attachments/files/16663258/KSP.zip)
(url)

**Comment by Nyxilo** [comment]
I have the same problem, which is quite annoying because it floods the ksp.log.
It automatically shows up in the flight scene, and the performance impact is pretty huge - I go from 8 FPS with RSE to 30 without, and I'm pretty sure it's because of this read/write spam in the log. 


Any hope of this ever being fixed?
From what I read online, this is appearing because something's lacking an initialization. 
https://docs.unity3d.com/6000.0/Documentation/Manual/null-reference-exception.html

**Comment by JonnyOThan** [comment]
@Nyxilo can you describe the precise steps to cause the bug to occur?

**Comment by Nyxilo** [comment]
I've installed RSE & RSE - Default using CKAN, then added [https://github.com/likeproblem/RSEFixes](url) for RWE compatibility, then launched a craft and looked into the console - as said on the forum thread, an error is thrown every 15 milisec ( https://forum.kerbalspaceprogram.com/topic/179579-1123-rocket-sound-enhancement-plugin-v096-070922-config-pack-v124-070922/?do=findComment&comment=4427733 ) 

I've tested with and without RSE fixes, the issue is still present. From what I can read in the log, the error shows up as soon as the scene is loaded, even before the craft.
Here it is :
https://1drv.ms/u/c/41573d9e9914a5dc/EbbD9pAvYblBtZ9zqJ1fz3MBSRMqzYOFmR8uJVPINMvzsA?e=dw9Om5

**Comment by JonnyOThan** [comment]
@Nyxilo i am 99% sure that I have done exactly that over and over again and I have not seen this error.  So I need you to be extremely precise in exactly what steps you took. What parts you connected, blow you launched the ship, etc.

**Comment by Nyxilo** [comment]
TL; DR : There's a conflict between RSE and ReStock, that can be triggered on a test craft using rigid attachement 🤨

First step : identifying the mods causing the exception :

1) Pure stock save
Aeris A3  no exception , one warning : cannot play disabled audio source

2) Installing RSE / RSE Default / Module Manager from CKAN....
=> Aeris A3 / Arrow E : no exception , same warning

3) Installing RWE / B9 Part Switch / Restock / Waterfall core from CKAN...
=> Aeris A3 / Arrow E / Osprey : no errors, same warning

4) Installing KSP Community Fixes from CKAN, optimization accepted during loading...
=> Osprey / Arrow E : no exception , same warning

5) Installing Ship Effects Continued from CKAN...
=> Aeris A3 / YMRV-25A (custom) : same warning, exception spam shows up only on the custom one.

6) Uninstalling every mod but RSE / RSE Default / Module Manager...
=> YMRV-25A (custom) : same warning, the exception appears only once at loading.

7) Reinstalling RWE / B9 Part Switch / Restock / Waterfall core...
=> YMRV-25A (custom) : same warning, the exception appears only once at loading.

8) Installing KSP Community Fixes from CKAN...
=> YMRV-25A (custom) : same warning, exception spam shows up.

9) Uninstalling RWE / B9 Part Switch / Restock / Waterfall core...
=> YMRV-25A (custom) : same warning, no exception.

**First conclusion : if RWE and RSE are installed together, one exception shows up. With KSPCF, this is looped.**


Second step : Identifying the part / module causing the exception.

Tests now done with RWE / RSE  and their dependencies from CKAN

1) Saving the craft, thus removing all usused modules from the OG modded save (FAR, RPM, etc).
=> Exception spam still shows up.

2) Removing parts after parts until exception spam stops.
=> Down to 20 parts, however any modification seems to remove the exception.

3) Removing KSPCF to have only one exception...
=> The loop still shows up ! KSPCF is then dismissed as an issue.

4) Removing RWE and installing ReStock...
=> The loop still shows up ! RWE is then dismissed as an issue.

**Second conclusion : The true error lies between RSE and Restock, and modifying any parts (moving, variant) removes the exception**

Third part : fiddling with parts settings 

1) Autostruts don't do anything. The loop is still present.
2) Rigid attachement seems to do something : If it's active on both the blades and the NB scanner, the loop activates. Else it doesn't.

**Finally, it seems like there is a conflict between Restock and RSE (at least only appears only when both are installed) that can be triggered while toggling the Rigid Attachement system. Any other modification to the test craft seems to remove permanently the error. Doing this to the OG craft doesn't remove the error tho.***

Here is the test craft :
https://1drv.ms/u/c/41573d9e9914a5dc/EapoHVfgJeBLkFLCOqxLYCwBGLmB7Ht8Ayw9-Ukw8MV1eQ?e=QcmrG2

And the KSP.log :
https://1drv.ms/u/c/41573d9e9914a5dc/EbbD9pAvYblBtZ9zqJ1fz3MBTh2Ko_E9Pe1Fu3iYbcRhjg?e=kDTVci

This seems to be a very strange bug, and I was not able to reproduce it with a .craft made from the ground up. My guess is that there is something weird happening in the .craft file that causes the exception loop, for whatever reason.

**Comment by JonnyOThan** [comment]
[RSE Debug craft.craft.txt](https://github.com/user-attachments/files/19535187/RSE.Debug.craft.craft.txt)

(test craft from above in case it disappears)

**Comment by Nyxilo** [comment]
Have you been able to reproduce the issue on your side ?

**Comment by JonnyOThan** [comment]
Haven’t tried; I’ll update this when I know anything.

**Comment by Wiget454** [comment]
I'm having the same exact NRE spam, but triggered under different? circumstances

 - I do have Restock, Restock+, and KSPCF installed

The spam starts for immediately after grappling any other vessel to this craft:

[MSRH Shipwright.craft.txt](https://github.com/user-attachments/files/21563128/MSRH.Shipwright.craft.txt)

(Minimum Mods needed to load: Restock, Restock+, Stockalike Station Parts Expansion Redux, Sandcastle)

The line(s) in the log:
```
[EXC 18:16:17.204] NullReferenceException: Object reference not set to an instance of an object
	RocketSoundEnhancement.RSE_PartAudioManager.LateUpdate () (at <80da5f92e59a4e1e8c62d932c2676817>:0)
	UnityEngine.DebugLogHandler:LogException(Exception, Object)
	ModuleManager.UnityLogHandle.InterceptLogHandler:LogException(Exception, Object)
	UnityEngine.Debug:CallOverridenDebugHandler(Exception, Object)
```

KSP.log:

[KSP.log](https://github.com/user-attachments/files/21563134/KSP.log)


As far as I am aware, rigid attachment is not enabled on that vessel or any of the vessels I am grappling it to.
